appmap/pytest.py

```diff
@@ -59,5 +60,10 @@ def pytest_pyfunc_call(pyfuncitem):
                 pyfuncitem.cls,
                 pyfuncitem.name,
                 method_id=pyfuncitem.originalname,
-                location=pyfuncitem.location):
-            yield
+                location=pyfuncitem.location) as metadata:
+            result = yield
+            try:
+                with testing_framework.collect_result_metadata(metadata):
+                    result.get_result()
+            except:  # pylint: disable=bare-except
+                pass  # exception got recorded in metadata
```

Comment:
Shouldn't this be allowing the exception to propagate?

Reply:
No, the exception is captured in the `result`. Hookwrappers shouldn't throw (unless there's a bug).